src/Microsoft.VisualStudio.ProjectSystem.Managed/Tree/ProjectImportsSubTreeProvider.cs

```diff
@@ -0,0 +1,288 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+
+namespace Microsoft.VisualStudio.Tree
+{
+    /// <summary>
+    /// Provides a top-level project sub-tree showing the tree of MSBuild project imports,
+    /// including props/targets from the SDK, targets, etc.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// To reduce clutter and memory usage, this tree is only visible (and populated) when
+    /// the "Show all Files" feature of Solution Explorer is enabled.
+    /// </para>
+    /// <para>
+    /// This feature can be enabled and disabled via the <c>ProjectImportsTree</c> project capability.
+    /// </para>
+    /// </remarks>
+    [Export(ExportContractNames.ProjectTreeProviders.PhysicalViewRootGraft, typeof(IProjectTreeProvider))]
+    [AppliesTo(Capability)]
+    internal sealed class ProjectImportsSubTreeProvider : ProjectTreeProviderBase, IProjectTreeProvider, IShowAllFilesProjectTreeProvider
+    {
+        public const string Capability = "ProjectImportsTree";
+        private const string RootNodeCaption = "Imports";
```

Comment:
Should this be in a resx for localization?

Reply:
Yes, thanks, fixed.